content/local/writer.go

```diff
@@ -130,12 +134,13 @@ func (w *writer) Commit(size int64, expected digest.Digest, opts ...content.Opt)
 //
 // To abandon a transaction completely, first call close then `Store.Remove` to
 // clean up the associated resources.
-func (cw *writer) Close() (err error) {
-	unlock(cw.ref)
-
-	if cw.fp != nil {
-		cw.fp.Sync()
-		return cw.fp.Close()
+func (w *writer) Close() (err error) {
+	if w.fp != nil {
```

Comment:
:man_shrugging: no idea, just like being safe :)

But in that case, maybe document in on the interface godoc. Or we can wait until someone find a good reason to implement that async behavior and report a bug ;-)